students/guillaume/quarter_2_project/leankitbackend/leankitbackend/history.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+from leankitbackend.time_import import files_to_import
+from leankitbackend.date_analysis import cvt_d, delta_now
+from leankitbackend.util import lst_to_xl, format_xl, lst_to_xl_format
+from leankitbackend.setting import init
+from leankitbackend.dataprocess import assigned, blocked
+from os import chdir, path
+from collections import defaultdict
+from prettytable import PrettyTable, MSWORD_FRIENDLY
+
+
+def weekly_subset(Cards_dict):
+    ddict = defaultdict(list)
+    for key, value in Cards_dict.items():
+        alpha = key.isocalendar()
+        key = '{}-{}'.format(alpha[0], alpha[1])
+        ddict[key].append(value)
+
+    for key, value in ddict.items():
+        tmp = max(value, key=lambda x: x.date)
+        ddict[key] = tmp
+
+    return dict(ddict)
+
+
+def projects(lst, Cards_dict, week_nb=None, Attr='projects'):
+
+    return {key: getattr(value, Attr) for key, value in Cards_dict.items()}
+
+
+def Active_projects(lst, Cards_dict, week_nb=None):
+    '''
+    Cards Dict is a dictionnary of leankit_cards
+    '''
+    return projects(lst, Cards_dict, week_nb, Attr='in_progress')
+
+
+def table_projects(projs):
+
+    def f(value):
+        tmp = list()
+        for item in value:
+
+            a = [item.title,
+                 cvt_d(item.createdOn),
+                 item.lane['title'],
+                 assigned(item),
+                 blocked(item),
+                 # cvt_d(item.movedOn),
+                 item.plannedFinish]
+            tmp.append(a)
+        return tmp
+
+    def g(value):
+        tmp = []
+        for item in value:
+            tmp.append(item[1:])
+            tmp.sort(key=lambda x: h(x))
```

Comment:
do you want to sort inside the loop? rather than after adding everything?

and you can jsut pass h in for the key function.

and I think this could be a list comp:

```
return sorted([item[1:] for item in value], key=h)
```